python/cuml/ensemble/randomforest.pyx

```diff
@@ -0,0 +1,410 @@
+                      int,
+                      int,
+                      int *,
+                      bool) except +
+
+    cdef void predict(cumlHandle& handle,
+                      rfClassifier[double] *,
+                      double *,
+                      int,
+                      int,
+                      int *,
+                      bool) except +
+
+    cdef RF_metrics cross_validate(cumlHandle& handle,
+                                   rfClassifier[float] *, float *, int *,
+                                   int, int, int *, bool)
+    cdef RF_metrics cross_validate(cumlHandle& handle,
+                                   rfClassifier[double] *, double *, int *,
+                                   int, int, int *, bool)
+
+    cdef RF_params set_rf_class_obj(int, int, float,
+                                    int, int, int,
+                                    bool, bool, int, int) except +
+
+
+cdef class RandomForest_impl():
+    """
+    Description and example code
+    split_algo = 0 for HIST, 1 for GLOBAL_QUANTILE and 3 for SPLIT_ALGO_END
+    """
+    cpdef object handle
+    cdef rfClassifier[float] *rf_classifier32
+    cdef rfClassifier[double] *rf_classifier64
+    cdef uintptr_t preds_ptr
+    cdef object n_estimators
+    cdef object max_depth
+    cdef object max_features
+    cdef object min_samples_split
+    cdef object n_bins
+    cdef object split_algo
+    cdef object min_rows_per_node
+    cdef object bootstrap
+    cdef object bootstrap_features
+    cdef object type_model
+    cdef object verbose
+    cdef object n_cols
+    cdef object rows_sample
+    cdef object max_leaves
+    cdef object gdf_datatype
+    cdef object stats
+    cdef object dtype
+
+    def __cinit__(self, n_estimators=10, max_depth=-1, handle=None,
+                  max_features=1.0, min_samples_split=2, n_bins=8,
+                  split_algo=0, min_rows_per_node=2,
+                  bootstrap=True, bootstrap_features=False,
+                  type_model="classifier", verbose=False,
+                  rows_sample=1.0, max_leaves=-1,
+                  gdf_datatype=None):
+
+        self.handle = handle
+        self.split_algo = split_algo
+        self.min_rows_per_node = min_rows_per_node
+        self.bootstrap_features = bootstrap_features
+        self.rows_sample = rows_sample
+        self.max_leaves = max_leaves
+        self.n_estimators = n_estimators
+        self.max_depth = max_depth
+        self.max_features = max_features
+        self.min_samples_split = min_samples_split
+        self.type_model = self._get_type(type_model)
+        self.bootstrap = bootstrap
+        self.verbose = False
+        self.n_bins = n_bins
+        self.rf_classifier32 = NULL
+        self.rf_classifier64 = NULL
+        self.n_cols = None
+
+    def _get_type(self, type_model):
+        if type_model == "classifier":
+            self.type_model = 1
+        else:
+            self.type_model = 0
+
+    """
+    TBD:
```

Comment:
Open a github issue and the correct way to make this comment is todo, not tbd

Reply:
Filed issue: https://github.com/rapidsai/cuml/issues/690